src/languageSupport/languages/sql/connection.ts

```diff
@@ -0,0 +1,240 @@
+import {ConnectionManager as AgnosticConnectionManager} from 'src/languageSupport/languages/agnostic/connection'
+
+// Types
+import {
+  DEFAULT_SQL_EDITOR_TEXT,
+  CompositionSelection,
+} from 'src/dataExplorer/context/persistance'
+import {
+  SelectableDurationTimeRange,
+  TagKeyValuePair,
+  TimeRange,
+} from 'src/types'
+
+// Utils
+import {DEFAULT_TIME_RANGE} from 'src/shared/constants/timeRanges'
+import {rangeToInterval as buildTimeRange} from 'src/shared/utils/sqlInterval'
+import {notify} from 'src/shared/actions/notifications'
+import {compositionEnded} from 'src/shared/copy/notifications'
+
+const castType = (value, _typ = null) => {
+  // Outstanding tech debt.
+  // For the TagValue selector (Qx builder), flux composition, and rest of codebase.
+  // It only supports tagValues of type string.
+  return `'${value}'`
+}
+
+const groupedTagValues = (
+  tagValues: TagKeyValuePair[]
+): {[key: string]: string[]} =>
+  tagValues.reduce(
+    (acc, {key, value}) => ({
+      ...acc,
+      [key]: (acc[key] || []).concat([value]),
+    }),
+    {}
+  )
+
+export class ConnectionManager extends AgnosticConnectionManager {
+  private _timeRange: TimeRange = DEFAULT_TIME_RANGE
+
+  _couldBeFromComposition(change) {
+    return change.forceMoveMarkers == true
+  }
+
+  _editorChangeIsWithinComposition(change) {
+    if (!this._compositionRange) {
+      return
+    }
+    const {
+      start: {line: startLine},
+      end: {line: endLine},
+    } = this._compositionRange
```

Comment:
```suggestion
    const {
      start: {line: startLine},
      end: {line: endLine},
    } as LspRange = this._compositionRange
```

It would be easier for future code maintainer to understand the shape of the object using`LspRange` or a defined type
https://github.com/influxdata/ui/blob/e4c590c99c1784a82e0f5184bd94dcd9b1db3b94/src/languageSupport/languages/flux/lsp/types/showMessage.ts#L53-L61

Reply:
the class prop itself is already typed as `protected _compositionRange: LspRange`